app/models/post.rb

```diff
@@ -2,4 +2,12 @@ class Post < ApplicationRecord
   belongs_to :user, class_name: 'User', foreign_key: 'author_id', counter_cache: true
   has_many :comments, class_name: 'Comment', foreign_key: true
   has_many :likes, class_name: 'Like', foreign_key: true
+
+  def most_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
```

Comment:
- [ ] OPTIONAL:  Helpful tip:  You can use the `after_save` rails callback in order to fire a method after each record is saved.  In this case, you can make the `update_post_counter` fire up every time a new post is saved.  Look:

```ruby
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Make sure to make this method private, and only this method!

Look:

```ruby
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  has_many :comments
  has_many :likes

  after_save :update_post_counter

  def most_recent_comments
    comments.order(created_at: :desc).limit(5)
  end

  private

  def update_post_counter
    user.increment!(:posts_count)
  end
end
```

I recommend you go ahead and implement this `after_save` for all your other counters: comments and likes. This will make your life easier in the upcoming milestones! 💪